### PR TITLE
sink(ticdc): fix daemon dbus process not stopped issue

### LINKED ISSUE #4926: daemon-dbus process will not be stopped after cdc is stopped
### What did you do?

ubuntu 20.04, 
run as the root user
remove DBUS_SESSION_BUS_ADDRESS env
run 
```
./cdc version
ps -ef | grep dbus-daemon
```
detail
https://asktug.com/t/topic/573263

### What did you expect to see?

no more dbus-daemon process is created, or created dbus-daemon is stopped

### What did you see instead?

one dbus-daemon process is created and not stopped.

### Versions of the cluster

TiCDC version (execute `cdc version`):

```console
Release Version: v5.3.0
Git Commit Hash: 20626babf21fc381d4364646c40dd84598533d66
Git Branch: heads/refs/tags/v5.3.0
UTC Build Time: 2021-11-22 10:37:02
Go Version: go version go1.16.4 linux/amd64
Failpoint Build: false
```

### PR DESCRIPTION
**Pull request by sdojjy**
### What problem does this PR solve?

Issue Number: close #4926

### What is changed and how it works?
 check DBUS_SESSION_BUS_ADDRESS first and then try to discovery it.
 if DBUS_SESSION_BUS_ADDRESS is found, do nothing, go-dbus will not create daemon-dbus,
if not, then create daemon-dbus, and set DBUS_SESSION_BUS_ADDRESS env and record the pid
kill the created process when cdc command is stopped

### Check List 

Tests 

 - Unit test
 - Integration test
 - Manual test 
 add DBUS_SESSION_BUS_ADDRESS env, check if the daemon process is stopped
 remove DBUS_SESSION_BUS_ADDRESS env, check if the daemon process is stopped

### Release note 

```release-note
`None`.
```

### CODE REVIEW

**Review by overvenus — COMMENTED** [review]
Could you patch godbus or keyring to stop spawning a dbus process?

File: cmd/cdc/hack/dbus_linux.go [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+
+// canDiscoverDbusSessionBusAddress check if we can discover an existing dbus session
+// and return the value of its DBUS_SESSION_BUS_ADDRESS.
+// It tries different techniques employed by different operating systems,
+// returning the first valid address it finds, or an empty string.
+//
+// * /run/user/<uid>/bus           if this exists, it *is* the bus socket. present on
+//                                 Ubuntu 18.04
+// * /run/user/<uid>/dbus-session: if this exists, it can be parsed for the bus
+//                                 address. present on Ubuntu 16.04
```
Comment (overvenus) [inline cmd/cdc/hack/dbus_linux.go:107]: What about other linux distribution?

Comment (sdojjy) [reply cmd/cdc/hack/dbus_linux.go:56]: no session bus address discovery on other Linux distribution according to the godbus source code.

File: cmd/cdc/main.go [hunk 2]
```diff
@@ -15,6 +15,7 @@ package main
 
 import (
 	_ "github.com/pingcap/tidb/types/parser_driver"
+	_ "github.com/pingcap/tiflow/cmd/cdc/hack"
 	"github.com/pingcap/tiflow/pkg/cmd"
```
[the remark this answers is not in the document]
Comment (sdojjy) [reply cmd/cdc/main.go:23]: done

**Review by overvenus — APPROVED** [review]
LGTM

File: cmd/cdc/hack/dbus_linux.go [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+		}
+		if runUserSessionDbusFile := path.Join(runtimeDirectory, "dbus-session"); fileExists(runUserSessionDbusFile) {
+			// if /run/user/<uid>/dbus-session exists, it's a
+			// text file // containing the address of the socket, e.g.:
+			// DBUS_SESSION_BUS_ADDRESS=unix:abstract=/tmp/dbus-E1c73yNqrG
+
+			if f, err := ioutil.ReadFile(runUserSessionDbusFile); err == nil {
+				fileContent := string(f)
+
+				prefix := "DBUS_SESSION_BUS_ADDRESS="
```
Comment (hi-rustin) [inline cmd/cdc/hack/dbus_linux.go:73]: Can we combine it with dbusSessionEnvName?

**Review by hi-rustin — APPROVED** [review]
This is a very hacky approach!

**Comment by ben1009** [comment]
so trick, another thing could we sent a issue to pulsar side also

**Pull request merged**